jbmc/src/java_bytecode/java_bytecode_parse_tree.cpp

```diff
@@ -99,23 +99,30 @@ void java_bytecode_parse_treet::annotationt::element_value_pairt::output(
   out << expr2java(value, ns);
 }
 
-bool java_bytecode_parse_treet::does_annotation_exist(
+/// Find an annotation given its name
+/// \param annotations: A vector of annotationt
+/// \param annotation_type_name: An irep_idt representing the name of the
+///   annotation class, e.g. java::java.lang.SuppressWarnings
+/// \return The first annotation with the given name in annotations if one
+///    exists, an empty optionalt otherwise.
+optionalt<java_bytecode_parse_treet::annotationt>
+java_bytecode_parse_treet::find_annotation(
   const annotationst &annotations,
   const irep_idt &annotation_type_name)
 {
-  return
-    std::find_if(
-      annotations.begin(),
-      annotations.end(),
-      [&annotation_type_name](const annotationt &annotation)
-      {
-        if(annotation.type.id() != ID_pointer)
-          return false;
-        typet type = annotation.type.subtype();
-        return
-          type.id() == ID_symbol
-          && to_symbol_type(type).get_identifier() == annotation_type_name;
-      }) != annotations.end();
+  const auto annotation_it = std::find_if(
+    annotations.begin(),
+    annotations.end(),
+    [&annotation_type_name](const annotationt &annotation) {
+      if(annotation.type.id() != ID_pointer)
+        return false;
+      const typet &type = annotation.type.subtype();
+      return type.id() == ID_symbol &&
+             to_symbol_type(type).get_identifier() == annotation_type_name;
+    });
+  if(annotation_it == annotations.end())
+    return {};
+  return *annotation_it;
```

Comment:
maye use `return ... ? ... : ... ;` here

Reply:
```return (annotation_it == annotations.end()) ? {} : *annotation_it;```
gives me an error, it doesn't like the `{}` in this notation. Is there anything I could replace it with? I've seen the two returns in other parts of the code as well...

Reply:
ah, right, this is interpreted differently here. One would have to use ```  return (annotation_it == annotations.end()) ? optionalt<java_bytecode_parse_treet::annotationt>()  : *annotation_it;``` which isn't really shorter. So better stay with the original code.

Reply:
Two returns is fine in that case. The two expressions in the ternary operator need to have the same type, but we cannot directly deduce the type of `{}`. Changing that to make it compile would probably be more complicated than the original `if(...) return {}; return ..;`.